Panic on failure to create environment repo

### Summary 

Reported in #jenkins-x-user:
```
Creating repository Riduidel/environment-snowcamp-2019-staging
Creating Git repository Riduidel/environment-snowcamp-2019-staging
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x30 pc=0x1372d00]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/kube.CreateEnvGitRepository(0x1, 0xc0001b9e30, 0x29, 0xc000cd1020, 0xc000031c00, 0xc0002dc000, 0xc000506898, 0x289048f, 0x3b, 0xc0001b9da0, ...)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/kube/env.go:283 +0x7d0
github.com/jenkins-x/jx/pkg/kube.CreateEnvironmentSurvey(0x2362601, 0xc0001b9e30, 0x29, 0xc000cd1020, 0xc000031c00, 0xc0002dc000, 0xc000506898, 0x289048f, 0x3b, 0x27fa2b1, ...)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/kube/env.go:236 +0x4f5
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateEnvOptions).Run(0xc0005066a0, 0x35, 0x0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_env.go:184 +0x4a2
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).Run(0xc0005061f8, 0x2b9bac0, 0xc00082a0a0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:1270 +0x4555
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterOptions).initAndInstall(0xc000506000, 0x27faa4f, 0x3, 0xc0007125a0, 0x10)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster.go:178 +0x7a
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterGKEOptions).createClusterGKE(0xc000506000, 0x27faa4f, 0x3)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:276 +0x9b2
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterGKEOptions).Run(0xc000506000, 0x0, 0x0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:121 +0x7f
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdCreateClusterGKE.func1(0xc0006bc500, 0xc0005c52c0, 0x1, 0xa)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:87 +0x6b
github.com/spf13/cobra.(*Command).execute(0xc0006bc500, 0xc0005c5180, 0xa, 0xa, 0xc0006bc500, 0xc0005c5180)
        /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2d3
github.com/spf13/cobra.(*Command).ExecuteC(0xc000524280, 0xc0007cd900, 0xc000760780, 0xc000793b80)
        /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x304
github.com/spf13/cobra.(*Command).Execute(0xc000524280, 0xc0006be940, 0x2baa5c0)
        /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x32
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc00077bf88, 0xc000054118)
        /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:17 +0xe3
main.main()
        /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:10 +0x29
```

The following line attempts to dereference the repo before checking whether the create errored:
https://github.com/jenkins-x/jx/blob/master/pkg/kube/env.go#L283

--- a/pkg/kube/env.go
+++ b/pkg/kube/env.go
@@ -278,15 +278,12 @@ func CreateEnvGitRepository(batchMode bool, authConfigSvc auth.ConfigService, de
 
 				if createRepo {
 					showURLEdit = false
-					r, p, err := createEnvironmentGitRepo(batchMode, authConfigSvc, data, forkEnvGitURL, envDir, gitRepoOptions, helmValues, prefix, git, in, out, errOut)
-					repo = r
-					url := r.CloneURL
-
+					var err error
+					repo, gitProvider, err = createEnvironmentGitRepo(batchMode, authConfigSvc, data, forkEnvGitURL, envDir, gitRepoOptions, helmValues, prefix, git, in, out, errOut)
 					if err != nil {
-						return repo, nil, err
+						return repo, gitProvider, err
 					}
-					gitProvider = p
-					data.Spec.Source.URL = url
+					data.Spec.Source.URL = repo.CloneURL
 				}
 			} else {
 				showURLEdit = true

